src/steps/fetch-users/index.ts

```diff
@@ -0,0 +1,218 @@
+            smsNumber: e.sms_number,
+            createdAt: e.created_at,
+            updatedAt: e.updated_at,
+            suspendedAt: e.suspended_at,
+            lastLogin: e.last_login,
+            delinquentAt: e.delinquent_at,
+            beta: e.beta,
+            defaultTeamId: e.default_team ? e.default_team.id : null,
+            defaultOrganizationId: e.default_organization
+              ? e.default_organization.id
+              : null,
+            identityProviderId: e.identity_provider
+              ? e.identity_provider.id
+              : null,
+          },
+          assign: {
+            _key: e.id,
+            _type: 'heroku_user',
+            _class: 'User',
+          },
+        },
+      });
+    });
+    await jobState.addEntities(userEntities);
+
+    const userToTeamRelationships = teamMembers
+      .filter((r) => fetchedUserIds.includes(r.user.id))
+      .map((r) => {
+        return createIntegrationRelationship({
+          _class: 'HAS',
+          from: {
+            _class: 'Team',
+            _type: 'heroku_team',
+            _key: r.teamId,
+          },
+          to: {
+            _class: 'User',
+            _type: 'heroku_user',
+            _key: r.user.id,
+          },
+          properties: {
+            role: r.role,
+          },
+        });
+      });
+    await jobState.addRelationships(userToTeamRelationships);
+
+    const userToAccountRelationships = accountMembers
+      .filter((r) => fetchedUserIds.includes(r.user.id))
+      .map((r) => {
+        return createIntegrationRelationship({
+          _class: 'HAS',
+          from: {
+            _class: 'Account',
+            _type: 'heroku_account',
+            _key: r.enterprise_account.id,
+          },
+          to: {
+            _class: 'User',
+            _type: 'heroku_user',
+            _key: r.user.id,
+          },
+          properties: {
+            permissions: JSON.stringify(r.permissions),
+          },
+        });
+      });
+    await jobState.addRelationships(userToAccountRelationships);
+  },
+};
+
+export default step;
+
+interface HerokuTeamMemberWithId extends HerokuTeamMember {
+  teamId: string;
+}
+
+async function getTeamMembers({
+  logger,
+  instance,
+  jobState,
+}: IntegrationStepExecutionContext): Promise<HerokuTeamMemberWithId[]> {
+  const heroku = new HerokuClient(instance.config);
+
+  logger.info('Retrieving heroku_teams from jobState...');
+  const teamIds = await getIdsFromJobState('heroku_team', jobState);
+  logger.info(`Retrieved ${teamIds.length} heroku_teams from jobState.`);
+
+  logger.info('Calling /teams/:teamId/members APIs...');
+  const teamsMembers = await Promise.all(
+    teamIds.map(async (id) => {
+      const teamMembers: HerokuTeamMember[] = await heroku.retryGet(
+        `/teams/${id}/members`,
+      );
+      return teamMembers.map((m) => {
+        return { ...m, teamId: id };
+      });
+    }),
+  );
```

Comment:
To clarify my other comment on this file, I think it would be nice to have `fetch-teams` and `fetch-team-members` steps, where `fetch-team-members` is a nice `iterateEntities({ _type: '...team' })` and fetch teams for each, without all this work to collect up all the team ids and then iterate those, within the `fetch-users` step.